sympy/printing/pretty/pretty.py

```diff
@@ -28,6 +28,8 @@
 pprint_use_unicode = pretty_use_unicode
 pprint_try_use_unicode = pretty_try_use_unicode
 
+#checks if expr has a function name
+has_function = False
```

Comment:
Using globals isn't safe here, it make printing non-threadsafe.

Slightly less bad would be to use `self.has_function`, but likely there's a better way to handle this